youtube_dl/extractor/duboku.py

```diff
@@ -0,0 +1,242 @@
+            'title': 'contains:预告片',
+            'season_number': 1,
+            'episode_number': 1,
+        },
+        'params': {
+            'skip_download': 'm3u8 download',
+        },
+    }]
+
+    _PLAYER_DATA_PATTERN = r'player_data\s*=\s*(\{\s*(.*)})\s*;?\s*</script'
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        temp = video_id.split('-')
+        series_id = temp[0]
+        season_id = temp[1]
+        episode_id = temp[2]
+
+        webpage_url = 'https://www.duboku.co/vodplay/%s.html' % video_id
+        webpage_html = self._download_webpage(webpage_url, video_id)
+
+        # extract video url
+
+        player_data = self._search_regex(
+            self._PLAYER_DATA_PATTERN, webpage_html, 'player_data')
+        player_data = self._parse_json(player_data, video_id, js_to_json)
+
+        # extract title
+
+        temp = get_elements_by_class('title', webpage_html)
+        series_title = None
+        title = None
+        for html in temp:
+            mobj = re.search(r'<a\s+.*>(.*)</a>', html)
+            if mobj:
+                href = extract_attributes(mobj.group(0)).get('href')
+                if href:
+                    mobj1 = re.search(r'/(\d+)\.html', href)
+                    if mobj1 and mobj1.group(1) == series_id:
+                        series_title = clean_html(mobj.group(0))
+                        series_title = re.sub(r'[\s\r\n\t]+', ' ', series_title)
+                        title = clean_html(html)
+                        title = re.sub(r'[\s\r\n\t]+', ' ', title)
+                        break
+
+        data_url = player_data.get('url')
+        if not data_url:
+            raise ExtractorError('Cannot find url in player_data')
+        data_from = player_data.get('from')
+
+        # if it is an embedded iframe, maybe it's an external source
+        if data_from == 'iframe':
+            # use _type url_transparent to retain the meaningful details
+            # of the video.
+            return {
+                '_type': 'url_transparent',
+                'url': smuggle_url(data_url, {'http_headers': {'Referer': webpage_url}}),
+                'id': video_id,
+                'title': title,
+                'series': series_title,
+                'season_number': int_or_none(season_id),
+                'season_id': season_id,
+                'episode_number': int_or_none(episode_id),
+                'episode_id': episode_id,
+            }
+
+        formats = self._extract_m3u8_formats(data_url, video_id, 'mp4')
+
+        return {
+            'id': video_id,
+            'title': title,
+            'series': series_title,
+            'season_number': int_or_none(season_id),
+            'season_id': season_id,
+            'episode_number': int_or_none(episode_id),
+            'episode_id': episode_id,
+            'formats': formats,
+            'http_headers': {'Referer': 'https://www.duboku.co/static/player/videojs.html'}
```

Comment:
Use `headers` as introduced above:
```suggestion
            'http_headers': headers,
```